azurerm/internal/services/containers/data_source_kubernetes_service_version.go

```diff
@@ -87,6 +102,11 @@ func dataSourceArmKubernetesServiceVersionsRead(d *schema.ResourceData, meta int
 					continue
 				}
 
+				if (isPreview) && (allowPreview == false) {
```

Comment:
we can simplify this too:

```suggestion
				if isPreview && allowPreview == false {
```